src/harness/core.ts

```diff
@@ -0,0 +1,671 @@
+                    }
+                    lineStart = pos;
+                    nonWhiteSpace = false;
+                    break;
+
+                // WhiteSpace
+                case 0x0009: // <TAB> tab
+                case 0x000b: // <VT> vertical tab
+                case 0x000c: // <FF> form feed
+                case 0x0020: // <SP> space
+                case 0x00a0: // <NBSP> no-break space
+                case 0xfeff: // <ZWNBSP> zero width no-break space
+                case 0x1680: // <USP> ogham space mark
+                case 0x2000: // <USP> en quad
+                case 0x2001: // <USP> em quad
+                case 0x2002: // <USP> en space
+                case 0x2003: // <USP> em space
+                case 0x2004: // <USP> three-per-em space
+                case 0x2005: // <USP> four-per-em space
+                case 0x2006: // <USP> six-per-em space
+                case 0x2007: // <USP> figure space
+                case 0x2008: // <USP> punctuation space
+                case 0x2009: // <USP> thin space
+                case 0x200a: // <USP> hair space
+                case 0x202f: // <USP> narrow no-break space
+                case 0x205f: // <USP> medium mathematical space
+                case 0x3000: // <USP> ideographic space
+                case 0x0085: // next-line (not strictly per spec, but used by the compiler)
+                    break;
+
+                default:
+                    nonWhiteSpace = true;
+                    break;
+            }
+        }
+        if (lineStarts) {
+            lineStarts.push(lineStart);
+        }
+        if (lines && (!removeEmptyElements || nonWhiteSpace)) {
+            lines.push(text.slice(lineStart, text.length));
+        }
+    }
+
+    export type LineStarts = ReadonlyArray<number>;
+
+    export interface LinesAndLineStarts {
+        readonly lines: ReadonlyArray<string>;
+        readonly lineStarts: LineStarts;
+    }
+
+    export function getLinesAndLineStarts(text: string): LinesAndLineStarts {
+        const lines: string[] = [];
+        const lineStarts: number[] = [];
+        splitLinesWorker(text, lineStarts, lines, /*removeEmptyElements*/ false);
+        return { lines, lineStarts };
+    }
+
+    export function splitLines(text: string, removeEmptyElements = false): string[] {
+        const lines: string[] = [];
+        splitLinesWorker(text, /*lineStarts*/ undefined, lines, removeEmptyElements);
+        return lines;
+    }
+
+    export function computeLineStarts(text: string): LineStarts {
+        const lineStarts: number[] = [];
+        splitLinesWorker(text, lineStarts, /*lines*/ undefined, /*removeEmptyElements*/ false);
+        return lineStarts;
+    }
+
+    //
+    // Cryptography
+    //
+
+    const H = new Uint32Array(5);
+    const W = new Uint8Array(80);
+    const B = new Uint8Array(64);
+    const BLOCK_SIZE = 64;
+
+    export function sha1(message: string): string {

```

Comment:
Since we added private packages for other things, this is something I'd rather see us use a private package for. Or, ideally, someone else's package, since it's only used for test.

Reply:
I can look into that in a follow-on PR.